Vagrantfile

```diff
@@ -0,0 +1,82 @@
+Vagrant.configure("2") do |config|
+  require 'etc'
+  config.vm.provider "virtualbox" do |vbox|
+    vbox.cpus = [1, Etc.nprocessors].max
+  end
+  config.vm.box = "centos/7"
+  config.vm.synced_folder ".", "/vagrant", disabled: true
+  config.vm.synced_folder ".", "/src/slirp4netns", type: "rsync"
+  config.vm.provision "shell",
+    inline: <<~'SHELL'
+      set -xeu
+      sysctl user.max_user_namespaces=65536
+
+      yum install -y \
+        epel-release \
+        https://repo.ius.io/ius-release-el7.rpm
+
+      yum install -y \
+        autoconf automake make gcc gperf libtool \
+        git-core meson ninja-build \
+        glib2-devel libcap-devel \
+        git-core libtool iproute iputils iperf3 nmap jq
+
+      cd /src
+      chown vagrant .
+
+      su vagrant -c '
+        set -xeu
+
+        git clone --depth=1 --no-checkout https://github.com/seccomp/libseccomp
+        git -C ./libseccomp fetch --tags --depth=1
+
+        git clone --depth=1 --no-checkout https://gitlab.freedesktop.org/slirp/libslirp.git
+        git -C ./libslirp fetch --tags --depth=1
+
+        touch ./build-and-test
+        chmod a+x ./build-and-test
+      '
+
+      cat > ./build-and-test <<'EOS'
```

Comment:
I have only included this static `build-and-test` script inline here because I wasn't sure if you'd prefer to have the `Vagrantfile` as self-contained as possible or if a separate script file would be okay/preferable.